src/gui/screen_menu_calibration.cpp

```diff
@@ -1,121 +1,31 @@
-    psmd->items[MI_Z_OFFSET].item.wi_spin_fl.range = zoffset_fl_range;
-}
+#include "screen_menu.hpp"
+#include "WindowMenuItems.hpp"
+#include "MItem_tools.hpp"
+#include "MItem_print.hpp"
 
-int8_t gui_marlin_G28_or_G29_in_progress() {
-    uint32_t cmd = marlin_command();
-    if ((cmd == MARLIN_CMD_G28) || (cmd == MARLIN_CMD_G29))
-        return -1;
-    else
-        return 0;
-}
+using Screen = screen_menu_data_t<EHeader::Off, EFooter::On, EHelp::Off, MI_RETURN, MI_WIZARD, MI_BABYSTEP, MI_AUTO_HOME, MI_MESH_BED,
+    MI_SELFTEST, MI_CALIB_FIRST>;
 
-int screen_menu_calibration_event(screen_t *screen, window_t *window, uint8_t event, void *param) {
-    if (screen_menu_event(screen, window, event, param))
-        return 1;
-    if ((event == WINDOW_EVENT_CHANGING) && ((int)param == MI_Z_OFFSET))
-        marlin_set_z_offset(psmd->items[MI_Z_OFFSET].item.wi_spin_fl.value);
-    else if ((event == WINDOW_EVENT_CHANGE) && ((int)param == MI_Z_OFFSET))
-        eeprom_set_var(EEVAR_ZOFFSET, marlin_get_var(MARLIN_VAR_Z_OFFSET));
-    else if (event == WINDOW_EVENT_CLICK) {
-        switch ((int)param) {
-        case MI_WIZARD:
-            wizard_run_complete();
-            break;
-        case MI_AUTO_HOME:
-            marlin_event_clr(MARLIN_EVT_CommandBegin);
-            marlin_gcode("G28");
-            while (!marlin_event_clr(MARLIN_EVT_CommandBegin))
-                marlin_client_loop();
-            gui_dlg_wait(gui_marlin_G28_or_G29_in_progress);
-            break;
-        case MI_MESH_BED:
-            if (!marlin_all_axes_homed()) {
-                marlin_event_clr(MARLIN_EVT_CommandBegin);
-                marlin_gcode("G28");
-                while (!marlin_event_clr(MARLIN_EVT_CommandBegin))
-                    marlin_client_loop();
-                gui_dlg_wait(gui_marlin_G28_or_G29_in_progress);
-            }
-            marlin_event_clr(MARLIN_EVT_CommandBegin);
-            marlin_gcode("G29");
-            while (!marlin_event_clr(MARLIN_EVT_CommandBegin))
-                marlin_client_loop();
-            gui_dlg_wait(gui_marlin_G28_or_G29_in_progress);
-            break;
-        case MI_SELFTEST:
-            wizard_run_selftest();
-            break;
-        case MI_CALIB_FIRST:
-            wizard_run_firstlay();
-            break;
-        }
-    }
-    return 0;
+static void init(screen_t *screen) {
+    marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_Z_OFFSET));
+    constexpr static const char *label = "Calibration";
+    Screen::Create(screen, label);
 }
 
 screen_t screen_menu_calibration = {
     0,
     0,
-    screen_menu_calibration_init,
-    screen_menu_done,
-    screen_menu_draw,
-    screen_menu_calibration_event,
-    sizeof(this_screen_data_t), //data_size
-    0,                          //pdata
+    init,
+    Screen::CDone,
+    Screen::CDraw,
+    Screen::CEvent,
+    sizeof(Screen), //data_size
+    0,              //pdata
 };
 
```

Comment:
`extern "C" screen_t *const get_scr_menu_calibration() { return &screen_menu_calibration; }`
warning: 'const' type qualifier on return type has no effect
and the same applies for other menus too

Reply:
cannot be resolved now, subject to future changes